packages/gcf-utils/src/gcf-utils.ts

```diff
@@ -360,58 +373,266 @@ export class GCFBootstrapper {
     };
   }
 
+  /**
+   * Entrypoint for handling all scheduled tasks.
+   *
+   * @param id {string} GitHub delivery GUID
+   * @param body {Scheduled} Scheduler params. May contain additional request
+   *   parameters besides the ones defined by the Scheduled type.
+   * @param signature
+   * @param wrapOptions
+   */
   private async handleScheduled(
     id: string,
     req: express.Request,
-    eventName: string,
     signature: string,
-    wrapOptions: WrapOptions | undefined
+    wrapOptions: WrapOptions
   ) {
     const body: Scheduled = this.parseRequestBody(req);
+    const cronType = body.cron_type ?? DEFAULT_CRON_TYPE;
+    if (cronType === 'global') {
+      await this.handleScheduledGlobal(id, body, signature);
+    } else if (cronType === 'installation') {
+      await this.handleScheduledInstallation(id, body, signature, wrapOptions);
+    } else {
+      await this.handleScheduledRepository(id, body, signature, wrapOptions);
+    }
+  }
+
+  /**
+   * Handle a scheduled tasks that should run once. Queues up a Cloud Task
+   * for the `schedule.global` event.
+   *
+   * @param id {string} GitHub delivery GUID
+   * @param body {Scheduled} Scheduler params. May contain additional request
+   *   parameters besides the ones defined by the Scheduled type.
+   * @param signature
+   */
+  private async handleScheduledGlobal(
+    id: string,
+    body: Scheduled,
+    signature: string
+  ) {
+    await this.enqueueTask({
+      id,
+      name: SCHEDULER_GLOBAL_EVENT_NAME,
+      signature,
+      body: JSON.stringify(body),
+    });
+  }
+
+  /**
+   * Async iterator over each installation for an app.
+   *
+   * See https://docs.github.com/en/rest/reference/apps#list-installations-for-the-authenticated-app
+   * @param wrapOptions {WrapOptions}
+   */
+  private async *eachInstallation(wrapOptions: WrapOptions) {
+    const octokit = await this.getAuthenticatedOctokit(undefined, wrapOptions);
+    const installationsPaginated = octokit.paginate.iterator(
+      octokit.apps.listInstallations
+    );
+    for await (const response of installationsPaginated) {
+      for (const installation of response.data) {
+        yield installation;
+      }
+    }
+  }
+
+  /**
+   * Async iterator over each repository for an app installation.
+   *
+   * See https://docs.github.com/en/rest/reference/apps#list-repositories-accessible-to-the-app-installation
+   * @param wrapOptions {WrapOptions}
+   */
+  private async *eachInstalledRepository(
```

Comment:
Love the use of async iterators, vs., our old approach of sometimes just grabbing the first page of results with octokit.